src/ml_board/backend/restful_api/data_access.py

```diff
@@ -103,50 +144,189 @@ def load_experiment_config(top_level_logging_path: str, grid_search_id: str, exp
             return response
 
         else:
-            raise InvalidPathError(f"File path {requested_full_path} is not safe.")
+            raise InvalidPathError(
+                f"File path {requested_full_path} is not safe.")
 
     def add_raw_config_to_grid_search(self, grid_search_id: str, config_name: str, config_file: RawTextFile):
-        requested_full_path = os.path.realpath(os.path.join(self.top_level_logging_path, str(grid_search_id), config_name))
+        """
+        Add Config for a Grid Search ID to event storage
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             config_name (str): Name of Configuration file
+             config_file (RawTextFile) : RawTextFile Object
+
+        """
+        requested_full_path = os.path.realpath(os.path.join(
+            self.top_level_logging_path, str(grid_search_id), config_name))
 
         if FileDataAccess.is_safe_path(base_dir=self.top_level_logging_path, requested_path=requested_full_path):
             os.makedirs(os.path.dirname(requested_full_path), exist_ok=True)
             with open(requested_full_path, "w") as fp:
                 fp.writelines(config_file.content)
         else:
-            raise InvalidPathError(f"File path {requested_full_path} is not safe.")
-
-    def get_raw_config_of_grid_search(self, grid_search_id: str, config_name: str):
-        raise NotImplementedError
+            raise InvalidPathError(
+                f"File path {requested_full_path} is not safe.")
 
     def add_config_to_experiment(self, grid_search_id: str, experiment_id: str, config_name: str, config: RawTextFile):
-        requested_full_path = os.path.realpath(os.path.join(self.top_level_logging_path, str(grid_search_id), str(experiment_id), config_name))
+        """
+        Add experiment config given the experiment ID & grid search ID to event storage.
+
+        :params: 
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             config_name (str): Name of Configuration file
+             config (RawTextFile) : RawTextFile Object
+
+        """
+        requested_full_path = os.path.realpath(os.path.join(
+            self.top_level_logging_path, str(grid_search_id), str(experiment_id), config_name))
 
         if FileDataAccess.is_safe_path(base_dir=self.top_level_logging_path, requested_path=requested_full_path):
             os.makedirs(os.path.dirname(requested_full_path), exist_ok=True)
             with open(requested_full_path, "w") as fp:
                 fp.writelines(config.content)
 
         else:
-            raise InvalidPathError(f"File path {requested_full_path} is not safe.")
+            raise InvalidPathError(
+                f"File path {requested_full_path} is not safe.")
+    
+    def get_grid_config_exec(self, grid_search_id: str, config_name: str):
```

Comment:
the function names were changed because there were same function names with different parameters and return type making it difficult to follow sometimes.
Reverted to name given.